TrackerUI/MainWindow.xaml.cs

```diff
@@ -0,0 +1,131 @@
+using System.Diagnostics;
+using System.IO;
+using nsTracker;
+using nsTask;
+
+namespace TrackerUI
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        private void Window_Loaded(object sender, RoutedEventArgs e)
+        {
+            StartTimer();
+            LoadComboboxItems();
+        }
+
+        private void LoadComboboxItems()
+        {
+            foreach (var item in Properties.Settings.Default.Combobox)
+            {
+                SelectionCombobox.Items.Add(item.ToUpper());
+            }
+        }
+
+        private void StartTimer()
+        {
+            DispatcherTimer timer = new DispatcherTimer();
+            timer.Interval = new TimeSpan(0, 0, 1);
+            timer.Tick += new EventHandler(UpdateTimerTick);
+            timer.Start();
+        }
+
+        private void UpdateTimerTick(object sender, EventArgs e)
+        {
+            CurrentTimeLabel.Content = DateTime.Now.ToString("dd/MM/yyyy HH:mm:ss");
+        }
+
+        private void AddButton_Click(object sender, RoutedEventArgs e)
+        {
+            if(SelectionCombobox.Items.Contains(AddTextBox.Text))
+                MessageBox.Show("You can not add the same task twice.");
+            else
+            {
+                SelectionCombobox.Items.Add(AddTextBox.Text.ToUpper());
+                AddTextBox.Clear();
+                UpdatePropertiesSettings();
+            }
+        }
+
+        private void UpdatePropertiesSettings()
+        {
+            Properties.Settings.Default.Combobox.Clear();
+            foreach (var item in SelectionCombobox.Items)
+            {
+                Properties.Settings.Default.Combobox.Add(item.ToString().ToUpper());
+            }
+            Properties.Settings.Default.Save();
+        }
+
+        private void DeleteButton_Click(object sender, RoutedEventArgs e)
+        {
+            SelectionCombobox.Items.Remove(SelectionCombobox.SelectedItem);
+            UpdatePropertiesSettings();
+        }
+
+        private List<Tasks> GetTasks()
+        {
+            for (int i = 0; i < Properties.Settings.Default.Combobox.Count; i++)
+            {
+                var comboboxItem = Properties.Settings.Default.Combobox[i];
+                var response = _tasks.Find(x => x.Name == comboboxItem);
+                if (response == null)
+                    _tasks.Add(new Tasks(comboboxItem.ToString().ToUpper()));
+            }
+            return _tasks;
+        }
+
+        private int GetSelection()
+        {
+            return SelectionCombobox.SelectedIndex;
+        }
+
+        private List<Tasks> _tasks = new List<Tasks>();
+
+        private void Button_Click(object sender, RoutedEventArgs e)
+        {
+            _tasks = GetTasks();
+            Tracker tracker = new Tracker(_tasks);
```

Comment:
This is not a good idea.
In this case, always after clicking on Start/Stop you are creating a new object from the beginning with all the tasks.

Reply:
Is not better to create the object Tracker in the beginning, when the UI starts, and then increase or decrease the list of tasks when you Add/Delete a task?